{"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["iOS"],"languages":["swift"],"tokens":[{"kind":"keyword","text":"typealias"},{"kind":"text","text":" "},{"text":"ResultCallback","kind":"identifier"},{"text":"<","kind":"text"},{"text":"Value","kind":"genericParameter"},{"text":"> = (","kind":"text"},{"identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/ResultValue","preciseIdentifier":"s:12TrustlessSDK11ResultValuea","kind":"typeIdentifier","text":"ResultValue"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","text":"Value","preciseIdentifier":"s:12TrustlessSDK14ResultCallbacka5Valuexmfp"},{"text":">) -> ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:s4Voida","text":"Void"}]}]},{"kind":"parameters","parameters":[{"name":"Value","content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"The type of the value that is expected on success."}]}]}]},{"content":[{"text":"Discussion","level":2,"anchor":"discussion","type":"heading"},{"type":"paragraph","inlineContent":[{"text":"This callback is used to return the outcome of asynchronous operations throughout the SDK. It allows for elegant handling of both success and failure cases without the need for cumbersome error-checking code.","type":"text"}]},{"type":"paragraph","inlineContent":[{"type":"text","text":"The "},{"type":"codeVoice","code":"ResultCallback"},{"type":"text","text":" takes a single parameter of type "},{"code":"ResultValue<Value>","type":"codeVoice"},{"type":"text","text":", enabling the caller to handle success and failure through the "},{"type":"codeVoice","code":"Result"},{"text":" typeâ€™s ","type":"text"},{"code":".success","type":"codeVoice"},{"text":" and ","type":"text"},{"code":".failure","type":"codeVoice"},{"text":" cases.","type":"text"}]}],"kind":"content"}],"variants":[{"paths":["\/documentation\/trustlesssdk\/resultcallback"],"traits":[{"interfaceLanguage":"swift"}]}],"abstract":[{"type":"text","text":"A typealias for a callback function that is invoked with a "},{"type":"codeVoice","code":"ResultValue"},{"type":"text","text":"."}],"kind":"symbol","sections":[],"hierarchy":{"paths":[["doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK"]]},"seeAlsoSections":[{"generated":true,"identifiers":["doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/ResultValue","doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/CancelCallback"],"title":"Request and Response"}],"schemaVersion":{"patch":0,"minor":3,"major":0},"identifier":{"url":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/ResultCallback","interfaceLanguage":"swift"},"metadata":{"roleHeading":"Type Alias","symbolKind":"typealias","fragments":[{"kind":"keyword","text":"typealias"},{"text":" ","kind":"text"},{"text":"ResultCallback","kind":"identifier"}],"modules":[{"name":"TrustlessSDK"}],"title":"ResultCallback","role":"symbol","externalID":"s:12TrustlessSDK14ResultCallbacka","navigatorTitle":[{"text":"ResultCallback","kind":"identifier"}]},"references":{"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/ResultCallback":{"kind":"symbol","fragments":[{"text":"typealias","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"ResultCallback"}],"title":"ResultCallback","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/ResultCallback","type":"topic","url":"\/documentation\/trustlesssdk\/resultcallback","abstract":[{"text":"A typealias for a callback function that is invoked with a ","type":"text"},{"code":"ResultValue","type":"codeVoice"},{"type":"text","text":"."}],"role":"symbol","navigatorTitle":[{"text":"ResultCallback","kind":"identifier"}]},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK":{"abstract":[],"kind":"symbol","role":"collection","url":"\/documentation\/trustlesssdk","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK","type":"topic","title":"TrustlessSDK"},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/CancelCallback":{"abstract":[{"text":"A typealias for a closure used to cancel an ongoing operation.","type":"text"}],"identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/CancelCallback","title":"CancelCallback","url":"\/documentation\/trustlesssdk\/cancelcallback","type":"topic","role":"symbol","kind":"symbol","navigatorTitle":[{"kind":"identifier","text":"CancelCallback"}],"fragments":[{"text":"typealias","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"CancelCallback"}]},"doc://com.staq.TrustlessSDK/documentation/TrustlessSDK/ResultValue":{"kind":"symbol","fragments":[{"kind":"keyword","text":"typealias"},{"text":" ","kind":"text"},{"text":"ResultValue","kind":"identifier"}],"title":"ResultValue","identifier":"doc:\/\/com.staq.TrustlessSDK\/documentation\/TrustlessSDK\/ResultValue","type":"topic","url":"\/documentation\/trustlesssdk\/resultvalue","abstract":[{"type":"text","text":"A typealias for the "},{"type":"codeVoice","code":"Result"},{"type":"text","text":" type specialized for the TrustlessSDK, where failures are represented by "},{"code":"TrustlessError","type":"codeVoice"},{"type":"text","text":"."}],"role":"symbol","navigatorTitle":[{"text":"ResultValue","kind":"identifier"}]}}}